src/python/CRABClient/ClientUtilities.py

```diff
@@ -232,23 +234,26 @@ def uploadlogfile(logger, proxyfilename, logfilename=None, logpath=None, instanc
         # Encode in ascii because old pycurl present in old CMSSW versions
         # doesn't support unicode.
         cacheurl = cacheurl['cacheSSL'].encode('ascii')
-        cacheurldict = {'endpoint': cacheurl, "pycurl": True}
 
-        ufc = UserFileCache(cacheurldict)
-        logger.debug("cacheURL: %s\nLog file name: %s" % (cacheurl, logfilename))
-        logger.info("Uploading log file...")
-        ufc.uploadLog(logpath, logfilename)
+        if 'S3' in cacheurl.upper():
+            #below line uploads log file to S3
+            uploadToS3(crabserver=crabserver, filepath=logpath, objecttype='clientlog', taskname=taskname, logger=logger)
+        else:
+            cacheurldict = {'endpoint': cacheurl, "pycurl": True}
+            ufc = UserFileCache(cacheurldict)
+            logger.debug("cacheURL: %s\nLog file name: %s" % (cacheurl, logfilename))
+            ufc.uploadLog(logpath, logfilename)
+            #logger.info("%sSuccess%s: Log file uploaded successfully." % (colors.GREEN, colors.NORMAL))
+            logfileurl = cacheurl + '/logfile?name='+str(logfilename)
+            if not username:
+                from CRABClient.UserUtilities import getUsername
+                username = getUsername(proxyFile=proxyfilename, logger=logger)
+            logfileurl += '&username='+str(username)
+            logger.info("Log file URL: %s" % (logfileurl))
         logger.info("%sSuccess%s: Log file uploaded successfully." % (colors.GREEN, colors.NORMAL))
-        logfileurl = cacheurl + '/logfile?name='+str(logfilename)
-        if not username:
-            from CRABClient.UserUtilities import getUsername
-            username = getUsername(proxyFile=proxyfilename, logger=logger)
-        logfileurl += '&username='+str(username)
-        logger.info("Log file URL: %s" % (logfileurl))
-        return  logfileurl
+
```

Comment:
I am puzzled by the removal of this block of code. Was there some bug ?

Reply:
I did not remove it, but moved it under `else` block.

Reply:
oops